jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpCookie.java

```diff
@@ -927,16 +957,150 @@ static String formatExpires(Instant expires)
     }
 
     /**
-     * <p>Parses the {@code Expires} attribute value
-     * (in RFC 1123 format) into an {@link Instant}.</p>
+     * @deprecated use {@link #parseCookieDate(String)} instead
+     */
+    @Deprecated(since = "12.0.9", forRemoval = true)
+    static Instant parseExpires(String expires)
+    {
+        return parseCookieDate(expires);
+    }
+
+    static Instant parseExpiresOld(String date)
+    {
+        return ZonedDateTime.parse(date, DateTimeFormatter.RFC_1123_DATE_TIME).toInstant();
+    }
+
+    /**
+     * <p>Parses a Cookie Date value (such as the {@code Expires} attribute) using algorithm
+     * specified in <a href="https://datatracker.ietf.org/doc/html/rfc6265#section-5.1.1">RFC6265: Section 5.1.1: Date</a>
+     * in into an {@link Instant}.</p>
      *
-     * @param expires an instant in the RFC 1123 string format
+     * @param date an instant in the string format supported by RFC6265 HTTP State Management
      * @return an {@link Instant} parsed from the given string
      */
-    static Instant parseExpires(String expires)
+    static Instant parseCookieDate(String date)
+    {
+        Objects.requireNonNull(date, "Expires string cannot be null");
+
+        int year = -1;
+        int month = -1;
+        int day = -1;
+        int hour = -1;
+        int minute = -1;
+        int second = -1;
+
+        try
+        {
+            int tokenCount = 0;
+            Iterator<String> tokenIter = Immutable.DATE_TOKENIZER.tokenize(date);
+            while (tokenIter.hasNext())
+            {
+                String token = tokenIter.next();
+                // ensure we don't exceed the number of expected tokens.
+                if (++tokenCount > 6)
+                {
+                    // This is a horribly bad syntax / format
+                    throw new IllegalStateException("Too many delimiters for a Date format");
+                }
+
+                if (token.isBlank())
+                    continue; // skip blank tokens
+
+                // RFC 6265 - Section 5.1.1 - Step 2.1 - time (00:00:00)
+                // if (hour == (-1) && Immutable.PATTERN_TIME.matcher(token).matches())
+                if (hour == (-1) && token.length() == 8 && token.charAt(2) == ':' && token.charAt(5) == ':')
+                {
+                    hour = StringUtil.toInt(token, 0);
```

Comment:
Set the hour last, so that if there is a problem with minutes or seconds, then hour remains -1

Reply:
ignore works fine